### ISSUE TITLE
[Bug]: `react/no-object-type-as-default-prop` is disabled when using a `ref` prop

### ISSUE THREAD

**Issue by JulienR1** [jsx-eslint/eslint-plugin-react#3767]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Brief description

The `react/no-object-type-as-default-prop` does not run when a component has a `ref` prop as a second argument.

### Show example of your code (as text format), add images/videos/gifs to help explain example

Both of these components should error because an object type is used as a default prop.
However, the second component (with a ref) does not error.

```jsx
const ErrorIsShown = ({ foo = {} }) => null;
const ErrorIsAbsent = ({ foo = {} }, ref) => null;
```

![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/12901664/895b91b6-0672-4051-97ee-321b77fa4f6a)

Link to repro: https://github.com/JulienR1/eslint-react-no-object-type-as-default-prop-repro

### What command(s) did you run to reproduce issue?

```bash
npm run lint
# OR
eslint . --ext js,jsx --report-unused-disable-directives --max-warnings 0
```

```js
// .eslintrc.cjs
module.exports = {
  root: true,
  env: { browser: true, es2020: true },
  extends: ["plugin:react/recommended"],
  ignorePatterns: ["dist", ".eslintrc.cjs"],
  parserOptions: { ecmaVersion: "latest", sourceType: "module" },
  rules: {
    "react/no-object-type-as-default-prop": "error",
    "react/prop-types": "off",
  },
};
```

### Brief description

Components with `ref` props should also generate errors when using default object type props.

![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/12901664/c06ba147-a7c0-410e-b7be-20607c5d6589)

### eslint-plugin-react version

v7.34.2

### eslint version

v8.57.0

### node version

v20.12.2

**Comment by JulienR1** [comment]
Actually, the issue does not come from the `ref` directly, it comes from having a second argument. If I use `forwardRef` as described in React's docs, I still don't get an error from eslint.

From what I can see, this behavior is caused by [this line](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/lib/rules/no-object-type-as-default-prop.js#L33) in the rule's implementation: having more than one argument in the function will not execute the rule.

**Comment by ljharb** [comment]
I see what you mean. Given that it's fine to have `, context` in a component, the rule should be triggered regardless of there being more than one param.

**Comment by JulienR1** [comment]
That is exactly it.
Do you want me to set up a PR fixing this?

**Comment by ljharb** [comment]
That’d be great, thanks!
